lib/serializer.ts

```diff
@@ -752,38 +747,40 @@ export interface Mapper {
   constraints?: MapperConstraints;
 }
 
+export type Mapper = BaseMapper | CompositeMapper | SequenceMapper | DictionaryMapper | EnumMapper;
+
 export interface PolymorphicDiscriminator {
   serializedName: string;
   clientName: string;
   [key: string]: string;
 }
 
-export interface CompositeMapper extends Mapper {
+export interface CompositeMapper extends BaseMapper {
   type: {
     name: "Composite";
     className: string;
-    modelProperties?: { [propertyName: string]: Mapper };
+    modelProperties: { [propertyName: string]: Mapper };
```

Comment:
This is a breaking change. Make sure you bump the minor version.

Reply:
Note that the removal of the index signature on BaseMapper type is what allows any of this to be checked in nested mappers at all--so the potential breaks go beyond this. I think the mappers we generate type-check against these interfaces before and after the recent generator changes, but I'll bump the minor version anyway.